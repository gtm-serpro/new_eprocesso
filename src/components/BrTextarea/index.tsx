import classNames from "classnames";
import uniqueId from "lodash/uniqueId";
import { ChangeEventHandler } from "react";

export type BrTextareaProps = {
  className?: string;
  id?: string;
  label?: string;
  info?: string;
  density?: "small" | "large";
  placeholder?: string;
  value?: string | number | ReadonlyArray<string> | undefined;
  onChange?: ChangeEventHandler<HTMLTextAreaElement>;
  readOnly?: boolean;
};

const BrTextarea = (props: BrTextareaProps) => {
  const { className, id, label, info, density, ...rest } = props;

  const textId = id || uniqueId("textarea-");

  const classes = classNames("br-textarea", density, className);

  return (
    <div className={classes}>
      {label && <label htmlFor={textId}>{label}</label>}
      <textarea id={textId} {...rest}></textarea>
      {info && <p className="text-base mt-1">{info}</p>}
    </div>
  );
};

export default BrTextarea;
