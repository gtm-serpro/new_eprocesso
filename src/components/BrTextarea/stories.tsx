import { Meta, Story } from "@storybook/react";

import BrTextarea, { BrTextareaProps } from ".";

export default {
  title: "Formulários/Textarea",
  component: BrTextarea,
} as Meta;

export const Simples: Story<BrTextareaProps> = (args) => (
  <BrTextarea {...args}>Conteúdo</BrTextarea>
);
