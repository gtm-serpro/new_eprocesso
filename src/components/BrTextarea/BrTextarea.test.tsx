import { render, screen } from "@testing-library/react";

import BrTextarea from ".";

describe("<BrTextarea />", () => {
  it("deveria renderizar o TextArea", () => {
    const { container } = render(<BrTextarea />);
    expect(container.firstChild).toHaveClass("br-textarea");
    expect(container.firstChild).toMatchSnapshot();
  });

  it("deveria renderizar o Label informado", () => {
    render(<BrTextarea label="label da textarea" />);
    screen.getByText("label da textarea");
  });

  it("deveria renderizar o placeholder informado", () => {
    render(<BrTextarea placeholder="placeholder da textarea" />);
    screen.getByPlaceholderText("placeholder da textarea");
  });
});
