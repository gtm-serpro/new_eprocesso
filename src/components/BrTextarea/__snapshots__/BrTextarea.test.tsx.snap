// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BrTextarea /> deveria renderizar o TextArea 1`] = `
<div
  class="br-textarea"
>
  <textarea
    id="textarea-1"
  />
</div>
`;
